### PR TITLE
feat: Add integration price test

### PR DESCRIPTION
**Pull request by zarazan**
- Add integration price test that runs the price-feeder using the sample config values with all functionality except submitting vote transactions. Then compares the computed prices against the prices from an external API (coinmarketcap.com) reporting a test error when the two numbers are outside of 0.1 standard deviation from each other.
- Add Github workflow for integration tests

Step one of: https://github.com/ojo-network/price-feeder/issues/105

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: tests/integration/coin_market_cap.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	q := url.Values{}
+	q.Add("symbol", strings.Join(symbols, ","))
+
+	apiKey := os.Getenv("COINMARKETCAP_API_KEY")
+	if apiKey == "" {
+		return nil, fmt.Errorf("COINMARKETCAP_API_KEY env var not set")
```
Comment (adamewozniak) [inline tests/integration/coin_market_cap.go:26]: Add a link in here to show where somebody can get it to run this test pls

File: config/config.go [hunk 2]
```diff
@@ -162,6 +162,33 @@ func (c Config) ProviderEndpointsMap() map[provider.Name]provider.Endpoint {
...
+		threshold, err := sdk.NewDecFromStr(deviation.Threshold)
+		if err != nil {
+			return nil, err
+		}
+		deviations[deviation.Base] = threshold
+	}
+	return deviations, nil
+}
+
+// ExpectedSymbols returns a slice of all unique base symbols from the config file.
```
Comment (adamewozniak) [inline config/config.go:179]:
```suggestion
// ExpectedSymbols returns a slice of all unique base symbols from the config object.
```

File: oracle/provider/websocket_controller.go [hunk 3]
```diff
@@ -193,9 +193,8 @@ func (conn *WebsocketConnection) SendJSON(msg interface{}) error {
 	defer conn.mtx.Unlock()
 
 	if conn.client == nil {
-		return fmt.Errorf("unable to send JSON on a closed connection")
+		return nil
```
Comment (adamewozniak) [inline oracle/provider/websocket_controller.go:196]: Should we at least log a debug warning here when the json client is closed?

File: tests/integration/util.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	variance := 0.0
+	for _, num := range numbers {
+		diff := num - mean
+		variance += diff * diff
+	}
+	variance /= float64(len(numbers))
+	return math.Sqrt(variance)
+}
+
+//lint:ignore U1000 helper function for integration tests
```
Comment (adamewozniak) [inline tests/integration/util.go:30]: We might want to configure lint to just not look in this directory instead

**Review by adamewozniak — APPROVED** [review]
🥳 
very excited to have this included

File: oracle/provider/websocket_controller.go (hunk 3 above)
Comment (zarazan) [reply oracle/provider/websocket_controller.go:196]: We already log an error when the socket is closed. This is always redundant. Either there already was a previous error or the websocket is being intentionally closed and we don't need to send any of the queued up messages.

Comment (adamewozniak) [reply oracle/provider/websocket_controller.go:196]: < 3

**Pull request merged**